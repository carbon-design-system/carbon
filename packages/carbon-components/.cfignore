*
!Staticfile
!demo/
demo/*
!demo/index.html
!demo/demo.js
!demo/demo.min.js
!demo/demo.min.js.map
!demo/demo.css
!demo/code/
!demo/code/**
!demo/component/
!demo/component/**
