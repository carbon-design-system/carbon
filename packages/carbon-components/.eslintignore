demo/*.css
demo/*.map
demo/*.js
demo/js/prism.js
demo/hot
!demo/index.js
dist
examples/codesandbox
tests/a11y-results
tests/coverage
/es
/umd
/scripts
/css
/scss
/html
/docs/js
node_modules
