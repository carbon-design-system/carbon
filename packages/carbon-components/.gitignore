# Ignore built files and static folder
demo/*.css
demo/*.map
demo/*.js
demo/*.html
demo/code
demo/component
demo/hot
!demo/index.js
!demo/components.js
dist
docs/js

# Test reports
tests/a11y-results
tests/coverage
tests/axe/allHtml

# Built files
/scripts
/html
/scss

# a11y testing
.aat.yml

# Generated Sass files
src/globals/scss/vendor
