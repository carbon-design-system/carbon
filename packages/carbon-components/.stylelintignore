# packages/components
demo/**
tests/**
css/**
scss/**

