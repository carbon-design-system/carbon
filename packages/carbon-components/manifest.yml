---
applications:
- name: carbon-dev-environment
  memory: 64M
  buildpack: https://github.com/cloudfoundry/staticfile-buildpack.git
routes:
  - route: vanilla.carbondesignsystem.com
