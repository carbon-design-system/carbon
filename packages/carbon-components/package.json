{
  "name": "carbon-components",
  "description": "The Carbon Design System is IBMâ€™s open-source design system for products and experiences.",
  "version": "11.23.1",
  "license": "Apache-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/carbon-design-system/carbon.git",
    "directory": "packages/components"
  },
  "bugs": "https://github.com/carbon-design-system/carbon/issues",
  "homepage": "https://www.carbondesignsystem.com/",
  "files": [
    "scss/**/*",
    "index.scss"
  ],
  "keywords": [
    "carbon",
    "component",
    "components",
    "css",
    "ibm",
    "library",
    "pattern",
    "patterns",
    "sass",
    "scss",
    "style guide",
    "carbon-design-system",
    "react"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "node tasks/build-styles.js",
    "postinstall": "carbon-telemetry collect --install"
  },
  "peerDependencies": {
    "sass": "^1.33.0"
  },
  "dependencies": {
    "@carbon/styles": "^1.23.1",
    "@carbon/telemetry": "0.1.0"
  },
  "devDependencies": {
    "@carbon/test-utils": "^10.26.0",
    "fs-extra": "^10.0.0",
    "rimraf": "^4.0.0",
    "sass": "^1.51.0"
  },
  "sideEffects": [
    "index.scss",
    "scss/**/*.scss",
    "scss/**/*.css"
  ]
}
