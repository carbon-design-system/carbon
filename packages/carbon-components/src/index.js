/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

// ====================//
// Imports and Exports //
// ====================//

// Base Elements & Components
// -------------
// - JavaScript classes for use with components and base-elements.
// - The following statements import classes from actual locations to
//   be consumed from this file instead of their actual locations.
export * from './globals/js/components';

export { default as settings } from './globals/js/settings';
