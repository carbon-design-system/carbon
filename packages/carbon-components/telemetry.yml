# yaml-language-server: $schema=https://unpkg.com/@ibm/telemetry-config-schema@v1/dist/config.schema.json

version: 1
projectId: fbc5603c-90ba-4d2f-b054-55108c9e06c0
endpoint: https://collector-prod.1am6wm210aow.us-south.codeengine.appdomain.cloud/v1/metrics
collect:
  npm:
    dependencies: null