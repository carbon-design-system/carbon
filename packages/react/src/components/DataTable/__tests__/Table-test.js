/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount, shallow } from 'enzyme';
import { Table } from '../';
import { settings } from 'carbon-components';

const { prefix } = settings;

describe('DataTable.Table', () => {
  it('should render', () => {
    const wrapper = mount(<Table className="custom-class" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should support disable zebra stripe', () => {
    const wrapper = shallow(<Table zebra={false} />);
    const table = wrapper.find(`.${prefix}--data-table`);
    expect(table.hasClass(`${prefix}--data-table--zebra`)).toBe(false);
  });

  it('should support enable sticky header', () => {
    const wrapper = shallow(<Table stickyHeader={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
