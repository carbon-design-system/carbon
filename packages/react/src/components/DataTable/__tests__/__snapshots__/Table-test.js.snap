// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.Table should render 1`] = `
<Table
  className="custom-class"
  isSortable={false}
  overflowMenuOnHover={true}
>
  <div
    className="bx--data-table-content"
  >
    <table
      className="bx--data-table custom-class bx--data-table--no-border"
    />
  </div>
</Table>
`;

exports[`DataTable.Table should support enable sticky header 1`] = `
<section
  className="bx--data-table_inner-container"
>
  <div
    className="bx--data-table-content"
  >
    <table
      className="bx--data-table bx--data-table--no-border bx--data-table--sticky-header"
    />
  </div>
</section>
`;
